test/extended/apiserver/tls.go

```diff
@@ -1,31 +1,360 @@
 package apiserver
 
 import (
+	"bytes"
+	"context"
 	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
 
 	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
 
+	configv1 "github.com/openshift/api/config/v1"
 	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/origin/test/extended/scheme"
 	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
 )
 
+var _ = g.Describe("[sig-api-machinery][Feature:APIServer]", func() {
```

Comment:
I haven’t checked in detail, but it looks like what we need – thanks! :) 

If you don’t mind, please add a comment explaining that this test only checks whether the components pick up the TLS profile, and that there is a periodic test which validates the platform more holistically by setting a TLS profile and running the e2e tests.